prelude.dx

```diff
@@ -63,7 +65,7 @@ def (.*)  (d:VSpace a) ?=> : Real -> a -> a = snd $ fromVSpace d
 
 @instance realVS : VSpace Real = MkVSpace realAdd (*)
 @instance tabVS  : VSpace a ?=> VSpace (n=>a) = MkVSpace tabAdd \s xs. for i. s .* xs.i
-
+@instance unitVS : VSpace Unit = MkVSpace unitAdd \s u. ()
```

Comment:
Hmm I'm not 100% sold on this one. I don't think we usually think of tuples as vector spaces, and unit is just an empty tuple. On the other hand, there are situations where we might want to treat `(Real, Real)` as a 2D vector... I'm kind of split on this one. wdyt @dougalm?

Reply:
Actually I'm all for it. Internally, we treat tuples as vector spaces when we do linearization. Tangent spaces have to be vector spaces, and we take `TangentSpaceOf (a & b)` to be `(TangentSpaceOf a & TangentSpaceOf b)`. The special case of the empty tuple is especially important because we use it for the tangent space of discrete types like Int. Sum types are another matter ...